api/src/clojure/org/akvo/flow_api/unilog/unilog.clj

```diff
@@ -147,3 +147,14 @@
                              (:form-instances-to-load events-2)))
           data-point-changed (doall (data-point/by-ids ds (:data-point-changed events-2)))]
       (assoc events-2 :form-instance-changed form-instances :data-point-changed data-point-changed))))
+
+
+(defn get-db-name [instance-id]
+  (str "u_" instance-id))
+
+(defn wrap-db-connection [handler unilog-db]
+  (fn [request]
+    (let [db-name (get-db-name (:instance-id request))
+          db-spec (-> unilog-db :spec (assoc :db-name db-name) event-log-spec)]
+      (with-open [connection (jdbc/get-connection  db-spec)]
```

Comment:
Suggestion: use the with-db-connection macro. The docs discourage using get-connection, plus I guess that using the macro means that we do not need to manually build the {:connection} map